### PR TITLE
Fix clippy warnings

### PR DESCRIPTION
**Pull request by russell-islam**
Signed-off-by: Muminul Islam <muislam@microsoft.com>

### CODE REVIEW

File: mshv-bindings/src/lib.rs [hunk 1]
```diff
@@ -6,7 +6,8 @@
 #![allow(non_upper_case_globals)]
 #![allow(non_camel_case_types)]
 #![cfg(target_arch = "x86_64")]
-
+#![allow(clippy::too_many_arguments)]
+#![allow(clippy::missing_safety_doc)]
```
Comment (liuw) [inline mshv-bindings/src/lib.rs]: Is this warning about lacking comments on unsafe blocks? If so, we should write that instead of ignoring this warning.

File: mshv-ioctls/src/ioctls/vm.rs [hunk 2]
```diff
@@ -88,7 +88,7 @@ impl VmFd {
         &self,
         interrupt_type: hv_interrupt_type,
         apic_id: u64,
-        vector: u32,
+        _vector: u32,
```
Comment (liuw) [inline mshv-ioctls/src/ioctls/vm.rs]: Why is this needed? `vector` is used.

File: mshv-ioctls/src/ioctls/vm.rs [hunk 3]
```diff
@@ -109,7 +109,7 @@ impl VmFd {
                 as_uint64: interrupt_type as u64 | ((control_flags as u64) << 32),
             },
             dest_addr: apic_id,
-            vector: vector,
+            vector: _vector,
```
Comment (liuw) [inline mshv-ioctls/src/ioctls/vm.rs]: You can instead just write `vector` here because the field name and variable name are the same.

File: mshv-ioctls/src/lib.rs [hunk 4]
```diff
@@ -7,7 +7,9 @@
 #![allow(non_upper_case_globals)]
 #![allow(non_camel_case_types)]
 #![cfg(target_arch = "x86_64")]
-
+#![allow(clippy::too_many_arguments)]
+#![allow(clippy::missing_safety_doc)]
```
Comment (liuw) [inline mshv-ioctls/src/lib.rs]: See my previous comment on this.

File: mshv-ioctls/src/ioctls/vm.rs (hunk 2 above)
Comment (russell-islam) [reply mshv-ioctls/src/ioctls/vm.rs]: Giving the chippy warnings. with the same name

Comment (liuw) [reply mshv-ioctls/src/ioctls/vm.rs]: Does it go away after adopting my suggestion below?

**Review by liuw — APPROVED** [review]

**Review by sameo — APPROVED** [review]
